assets/js/translator.js

```diff
@@ -646,8 +657,72 @@ function populateTranslationList() {
             }
         }
 
-        srcLangs = srcLangs.sort(compareLangCodes);
-        dstLangs = dstLangs.sort(function (a, b) {
+        var langsOnly = [];
+        var variantsOnly = [];
+
+        for(i = 0; i < srcLangs.length; i++) {
+            if(srcLangs[i].indexOf('_') !== -1) {
+                variantsOnly.push(srcLangs[i]);
+            }
+            else {
+                langsOnly.push(srcLangs[i]);
+            }
+        }
+
+        var langHead = 0;
+        var variantHead = 0;
+        var numLangs = srcLangs.length;
+        srcLangs = [];
+
+        langsOnly = langsOnly.sort(compareLangCodes);
+        variantsOnly = variantsOnly.sort(compareLangCodes);
+
+        while((langHead + variantHead) < numLangs) {
```

Comment:
I'm a little confused by this code. If the point of it is to get the order correct, I think `compareLangCodes` might already do that? Also, it should be extracted into a function since its used twice and is pretty long ;)

Reply:
(I didn't read the code super carefully so I could be misunderstand the intent)